lib/sender.js

```diff
@@ -9,14 +9,17 @@ const { mask: applyMask, toBuffer } = require('./buffer-util');
 
 const mask = Buffer.alloc(4);
 
+
+/** @typedef {import('net').Socket} Socket */
```

Comment:
This could also be a `tls.Socket`. Also, no TypeScript-specific syntax please.

Reply:
i tried tls.Socket, didn't work so well... it could not figure out the correct NodeJS namespace
also typedef is a jsdoc thing

Reply:
https://stackoverflow.com/questions/49836644/how-to-import-a-typedef-from-one-file-to-another-in-jsdoc-using-node-js

Reply:
From the comments:

> This is TypeScript-specific syntax. Vote for github.com/jsdoc/jsdoc/issues/1645.

Reply:
oh, look at that... 
anyway another way could be to import a unused variable

```js
var http = require('http')

* @param {http.Server}
function foo() {
}
```

Reply:
That's acceptable. However we might need to add an eslint comment for unused variables.